sklearn/ensemble/_gb.py

```diff
@@ -476,6 +531,11 @@ def fit(self, X, y, sample_weight=None, monitor=None):
             )
 
         # if not warmstart - clear the estimator state
+        check_scalar(
+            self.warm_start,
+            name="warm_start",
+            target_type=bool,
+        )
```

Comment:
This works:

```python
check_scalar(self.warm_start, "warm_start", bool, min_val=0, max_val=1)
```

This would break code which use `warm_start=1`, so to be safe:

```python
check_scalar(self.warm_start, "warm_start", numbers.Integral, min_val=0, max_val=1)
```